app/main.cpp

```diff
@@ -137,7 +138,12 @@ int main( int argc, char **argv ) {
 	constexpr int_data_type usecs {2000000}; // microseconds
 	std::cout << "Displaying projections during progressive analysis:" << std::endl; 
 	while(lsf.calcNextProjection()) {
-		std::cout << '\r' << "a: " << lsf.getProja() << " b: " << lsf.getProjb() << std::flush;
+		auto start = std::chrono::system_clock::now();
+		std::cout << '\r' << "a: " << lsf.getProja() << " b: " << lsf.getProjb();
```

Comment:
I hadn't noticed that I left some profiling code in main.cpp.  app/profiling.cpp is where the profiling is intended to be done.  I did a few experiments in main.cpp first but forgot to remove the evidence!

main.cpp is intended as my personal playground to poke around with the classes/functions and convince myself they work as intended.  Primarily I'm using it to display results of analysis, giving me a chance to cross-reference the results with some manual/spreadsheet calculations.

My intention going forward with profiling.cpp is to write data to disk instead of printing to console, however just to demonstrate that it is actually profiling, and that the values appear to be reasonable, I was printing them to console for now.  I'll be modifying this in order to generate plots of profiling data this week.

Does this approach seem good to you?

Reply:
Yep, sounds good! In fact, if you take the `struct projection` suggestion described below, you will already have overloaded the stream operator for printing this out, and you'd just need to stream to the file instead of `std::cout`; e.g.,:
```
ofstream file("filename.txt");
file << proj << std::endl;
```